src/non-standalone/Dialog/ImageDialog.tsx

```diff
@@ -0,0 +1,247 @@
+			changeRef.current.dispatchEvent(event);
+		},
+		[convertImagesTo, downscale, localuploadedImages, setImages]
+	);
+	// upload click handler
+	const handleUpload = useCallback(() => {
+		const elem = document.createElement("input");
+		elem.type = "file";
+		elem.accept = "image/*";
+		elem.multiple = false;
+		elem.addEventListener("change", () => {
+			const files = elem.files;
+			if (!files) return;
+			const file = files[0];
+			if (!file) return;
+			void processFile(file);
+		});
+		elem.click();
+	}, [processFile]);
+	const changePrimaryImage = useCallback(
+		(index) => {
+			const newUpdatedImages = localuploadedImages.map((ele, ind) => {
+				if (index === ind) {
+					ele.setPrimary = !ele.setPrimary;
+				} else {
+					ele.setPrimary = false;
+				}
+				return ele;
+			});
+			setImages(newUpdatedImages);
+			onPrimarySelected(newUpdatedImages);
+		},
+		[localuploadedImages, onPrimarySelected]
+	);
+	const removeImageHandler = useCallback(
+		(index) => {
+			localuploadedImages.splice(index, 1);
+			setImages([...localuploadedImages]);
+			onUpdateImages(localuploadedImages);
+		},
+		[localuploadedImages, setImages, onUpdateImages]
+	);
+	const handleDrop = useCallback(
+		async (evt: React.DragEvent<HTMLDivElement>) => {
+			if (readOnly) return;
+
+			evt.preventDefault();
+
+			await processFile(evt.dataTransfer?.files[0]);
+		},
+		[readOnly, processFile]
+	);
+
+	const handleDragOver = useCallback(
+		(evt: React.DragEvent<HTMLDivElement>) => {
+			if (readOnly) return;
+
+			evt.preventDefault();
+		},
+		[readOnly]
+	);
+	return (
+		<Dialog
+			onClose={closeDialog}
+			maxWidth="lg"
+			fullWidth
+			disableBackdropClick
+			open={true}
+		>
+			<MuiDialogTitle disableTypography>
+				<IconButton
+					aria-label="close"
+					className={classes.closeButton}
+					onClick={closeDialog}
+				>
+					<Close />
+				</IconButton>
+			</MuiDialogTitle>
+			<DialogContent>
+				<Grid item>
+					<Grid container spacing={2}>
+						{localuploadedImages.map((image, i) => (
+							<Grid key={i}>
+								<Grid item xs className={classes.layout}>
+									{!image.setPrimary && (
+										<Cancel
+											onClick={() => removeImageHandler(i)}
+											className={classes.closeIcon}
+										/>
+									)}
+									<img
+										src={image.src}
+										alt={props.name}
+										className={classes.addPhotoContainer}
+									/>
+								</Grid>
+								<Grid container alignItems="center" style={{ margin: "30px" }}>
+									<Star
+										color={image.setPrimary ? "primary" : "disabled"}
+										onClick={() => changePrimaryImage(i)}
+									/>
+									{setPrimaryLabel}
+								</Grid>
+							</Grid>
+						))}
+						{!readOnly && (
+							<Grid className={classes.layoutAdd}>
+								<Grid item xs onDrop={handleDrop} onDragOver={handleDragOver}>
+									<div className={classes.addPhotoContainer}>
+										<AddAPhoto
+											style={{
```

Comment:
no inline styles please. you can set multiple classes using className={`${classes.addPhoto} ${classes.addPhotoIcon}`} if needed